invokeai/backend/stable_diffusion/extensions/freeu.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+from contextlib import contextmanager
+from typing import TYPE_CHECKING, Dict, Optional
+
+import torch
+from diffusers import UNet2DConditionModel
+
+from invokeai.backend.stable_diffusion.extensions.base import ExtensionBase
+
+if TYPE_CHECKING:
+    from invokeai.app.shared.models import FreeUConfig
+
+
+class FreeUExt(ExtensionBase):
+    def __init__(
+        self,
+        freeu_config: Optional[FreeUConfig],
```

Comment:
Currently, `FreeUExt` is only initialized if a valid freeu_config is available. So, this parameter can be made non-optional and we can remove the `if self.freeu_config is not None:` later in this class.